osg_configure/configure_modules/localsettings.py

```diff
@@ -47,6 +49,12 @@ def parse_configuration(self, configuration):
             if name in configuration.defaults():
                 self.log("%s is a default, skipping" % (name))
                 continue
+            # Validate name because it will be used as the name of an env var
+            if not re.match(r"[A-Za-z_][A-Za-z0-9_]*$", name):
+                raise SettingError("Invalid variable name " + name)
+            # Values are already double-quoted so adding quotes would be a syntax error
+            if value.startswith('"') or value.rstrip().endswith('"'):
```

Comment:
Is it the case then that `value` gets automatically `.lstrip()`ped?

Reply:
Perhaps more generally, is a `"` ever supposed to be allowed in `value`?
If not, then perhaps the check could just be `if '"' in value:`

Reply:
Yep, it's automatically lstripped. These will get turned into shell variable assignments so I want to leave the user the option to add quotes in there (they'd have to backslash them) but surrounding the whole thing with quotes is almost always a mistake.